Comments: design tweaks

drw158 opened the pull request:
A handful of tweaks including:

- Destructive buttons in bulk edit are now `scary`
- Pending comments are now yellow
- Updated action button colors, including :hover
- Increased contrast
- Removed `fill` CSS that wasn't necessary

To test:

- Check bulk edit buttons in multiple filters and confirm that the destructive actions are `scary`
- Check multiple filters and make sure pending comments are yellow

Before

![screen shot 2017-06-13 at 4 21 12 pm](https://user-images.githubusercontent.com/618551/27105213-5d6c7cf4-5054-11e7-9a03-34e8357a65c7.png)


After 

![screen shot 2017-06-13 at 4 20 00 pm](https://user-images.githubusercontent.com/618551/27105217-62261f52-5054-11e7-88c2-2427e4362dc5.png)

(screenshots are showing hover on the Spam button)

cc @Copons

gwwar commented on client/blocks/comment-detail/style.scss:
```diff
@@ -233,7 +254,16 @@
+}
+
+// If the comment is pending and expanded, color it yellow.
+.comment-detail:not( .is-approved ) .comment-detail__comment {
```
We can probably group this with the rule above:
```
.comment-detail.card:not( .is-approved ):not( .is-expanded ) .comment-detail__header,
comment-detail:not( .is-approved ) .comment-detail__comment {
    background: rgba( $alert-yellow, 0.085 );
}
```

Copons commented on client/blocks/comment-detail/style.scss:
```diff
@@ -17,6 +17,14 @@
 	}
 
+	// If the comment is pending and collapsed, color it yellow.
+	&:not( .is-approved ):not( .is-expanded ) {
```
This is caused by my poor CSS classes design, but I'm actually not really into this whole `:not` part.

I'd rather add a `.is-pending` and `.is-closed`(-contract? -shrunk?) [here](https://github.com/Automattic/wp-calypso/blob/master/client/blocks/comment-detail/index.jsx#L132-L140), and update the stylesheet accordingly.

Copons commented:
Re: the comment content border not being as long as the container: it should be an easy fix.

- Locate `.comment-detail__comment-body` and replace the padding to `padding: 16px`.

- Locate `.comment-detail.is-expanded .comment-detail__author-preview` and remove the `padding-bottom`.

- Add something like:
```css
.comment-detail__comment-body p:last-child {
  margin-bottom: 0;
}
```
Maybe we could even be more generous and go with:
`.comment-detail__comment-body > :last-child` and target _any_ last child, not only paragraphs?

Copons commented on client/blocks/comment-detail/comment-detail-actions.jsx:
```diff
@@ -76,8 +76,8 @@ export const CommentDetailActions = ( {
 						isSpam
-							? translate( 'Spam' )
-							: translate( 'Mark as Spam' )
+							? translate( 'Spammed' )
```
This sound very odd to me, I've always seen this used as an active verb, like "I've spammed (sent spam email) all the contacts in my address book".
Not that it really matters, since this label shouldn't ever be visible (there is no spam icon/action in the spam list).

drw158 commented:
Thanks for the review! Ready for another review. I have

- added collapsed and unapproved class names
- extended the left border all the way to the bottom
- removed pending expanded comment styles for now, until an alternate design is finished, to address Kerry's concerns above.

As a result of adding `is-unapproved` class, this removes the yellow style from trashed and spammed comments in their respective filters. Before, it targeted any comment that was not `is-approved`. I think this is ok, and might actually be better.

drw158 commented:
@gwwar Those are good points. The yellow border in the list view works in grayscale, because it adds a border. In the comment though, in just changes the border color, so not the best in terms of accessibility.

It might be hard to remember what filter you are in so I think maybe adding a label to each expanded comment would help, and also solve the accessibility problem.

Here are some ideas

![screen shot 2017-06-15 at 1 49 21 pm](https://user-images.githubusercontent.com/618551/27197218-0bbae8a2-51d3-11e7-8820-2d6418f0f31e.png)

gwwar commented:
Nice! @drw158 I like the status labels of 1 and 2. Should we also ask some other folks for 🤔 ?

roundhill commented:
> Nice! @drw158 I like the status labels of 1 and 2. Should we also ask some other folks for 🤔 ?

I really like the label 👍

Copons commented:
Labels look great!

Not sure if we'd still need the yellow border/background though.
I mean, the third draft there looks somewhat more "dangerous" than the others.
Not sure if it's the intended behaviour, though.
Like, is pending more "CTA" than spam or trash?

drw158 commented:
@Copons That's a good way to look at it. Pending is more pressing, so it should look more "CTA". I think the yellow styling of the list item will be good enough though. No need for the yellow expanded view.

One thing to point out though is that people like the yellow comment background in the mobile apps in notifications. This would be a slight deviation. But, no need for labels in notifications since you can only see pending/approved.

I say we go ahead and merge this, which removes the yellow expanded style, and we can add the labels later. We can ask for some more 🤔 on that PR

gwwar commented:
> I say we go ahead and merge this, which removes the yellow expanded style, and we can add the labels later. We can ask for some more 🤔 on that PR

Sounds good to me! Feel free to 🚢
